scripts/javascript.js

```diff
@@ -0,0 +1,163 @@
+
+/* 
+--------------
+FETCH FUNCTION
+--------------
+*/
+const goFetch = function(fullURL) {
+  fetch(fullURL) // by default fetch makes a GET request
+    .then(function(response) {
+      return response.json();
+    })
+    .then(function(body) {
+      displayDataOnPage(body);
+    })
+    .catch(function(error) {
+      displayErrorToUser("Server failed to return data");
+    });
+};
+
+/*
+--------------------
+DISPLAY DATA ON PAGE
+--------------------
+*/
+const newsLayout = newsitem => {
+  // fallbacks or empty if data is null
+  const imageurl =
+    newsitem.urlToImage !== null
+      ? `${newsitem.urlToImage}`
+      : "https://placeholdit.co//i/400x400?&bg=808069&fc=eee8cd&text=Project Image";
+  const source =
+    newsitem.source.name !== null
+      ? `<span class="${newsitem.source.name}"></span>`
+      : "";
+  const url = newsitem.url !== null ? `${newsitem.url}` : "#";
+  const title = newsitem.title !== null ? `${newsitem.title}` : "News article";
+  const description =
+    newsitem.description !== null ? `<h3>${newsitem.description}</h3>` : "";
+  const content =
+    newsitem.content !== null
+      ? `<p>${newsitem.content.split("[+")[0]}</p>`
+      : "";
+
+  return `
+    <a href="${url}" class="news--article-link">
+    <figure class="news--article-image" style="background-image: url(${imageurl});" ></figure>
+    ${source}
+    </a>
+    <section class="news--article-content">
+    <header><h2>${title}</h2></header>
+    ${description}
+    ${content}
+    <p><a href="${url}" title="Visit news article: ${title}">Read full article</a></p>
+   <input class="redcard-checkbox" type="checkbox" name="ratenews"> <label>Show article the Red Card!</label>
+    </section>`;
+};
+
+// TOFIX cleared on refresh, due to reset on displayDateOnPage?
+function redCard() {
+  const redcardCheckbox = document.querySelectorAll(".redcard-checkbox");
+
+  return redcardCheckbox.forEach(item => {
+    item.addEventListener("change", function(event) {
+      event.target.checked
+        ? (event.target.parentNode.style.backgroundColor = "red")
+        : (event.target.parentNode.style.backgroundColor = "");
+    });
+  });
+}
+
+function displayDataOnPage(newsStories) {
+  parentNode.innerHTML = "";
+
+  const newsArray = newsStories.articles;
+  // add news blocks (as articles)
+  newsArray.forEach(function(newsitem) {
+    const node = document.createElement("article");
+    node.innerHTML = newsLayout(newsitem);
+    parentNode.appendChild(node);
+  });
+  //
+  redCard();
+}
+
+/*
+------------------
+GENERATE FETCH URL
+------------------
+*/
+
+// news sources object
+let publicationList = {
+  "daily-mail": false,
+  mirror: false,
+  metro: false
+};
+
+const generateFetchURL = function(publicationList) {
+  // default url array
+  const defaultArray = ["daily-mail", "mirror", "metro"];
+
+  // create an array from object using key values
+  let publicationArray = Object.keys(publicationList);
```

Comment:
if variables will not be re-assigned to in future they should be declared using `const`